components/test-page-setup.js

```diff
@@ -165,7 +186,12 @@ function setVis(stateVar, str) {
         return ""
     }
 }
-
+let totalIteration = 0;
```

Comment:
why do we need this variable, separate from the `iterations.value`?

Reply:
SInce react hook can only be called inside a body of the function and I want to export the change, it's probably easier to use a variable to catch the change of the iteration. I'll leave this for now and move on to the rest of your comments